src/ethereum/shanghai/fork.py

```diff
@@ -617,8 +617,7 @@ def process_transaction(
         
     floor = tokens_in_calldata * TOTAL_COST_FLOOR_PER_TOKEN + TX_BASE_COST
     if gas_used < floor:
-        if floor > tx.gas:
-            raise 
+        ensure(floor > tx.gas, InvalidBlock) 
```

Comment:
This may be backwards? It is now saying `floor` must be greater than `tx.gas` or else it'll raise `InvalidBlock`.

Reply:
Right. Fixed it - thanks!